### DISCUSSION TITLE
Here to learn

### DISCUSSION THREAD

**Discussion by ParrhesiaJoe** [cssbattle/forum#361] [Ideas]
I signed up for a plus account because I planned on using this site to refine my CSS skills.

1. The goal of the battle seems to promote creating unprofessional code. I just noticed that I can reduce my score by removing white space, and by writing code that would make a code reviewer reject it (no quotes, unclosed tags). Can't you just run a minify on the output and compare the scores after minify? Also, it seems completely reasonable to require that the code is well formed. When I read the "tips" and it said, basically, "Don't close your tags", my brain started to bleed.

2. More importantly, I want to see solutions. I can't improve my code if I don't see what other people are doing. Even if you set a 2 week lag time or an "opt-in" for people to show their code, that would help me out immensely. There is nothing here for me as a learner that I can't just get from Codepen or VSCode.

Good idea for a site... but...

I will be cancelling my subscription next month.

**Comment by chinchang** [comment]
Hello @ParrhesiaJoe we sent you a mail sometime back. Please let us know there whenever you get a chance to go through it. 

About your above feedback - you are totally right! CSSBattle, on terms of code, doesn't at all relate to the code you would write in the professional world. And that is because the platform is based on a specific type of programming challenge called ["Code Golfing"](https://en.wikipedia.org/wiki/Code_golf) wherein you do a certain task in the least number of characters, thus requiring a lot of quirks, hacks and clever code. Which is as you mentioned totally opposite to the professional world. So, for learning to code, CSSBattle is not the right place.

That said, CSSBattle does help in learning CSS a lot. If you did search on Twitter for "css battle" you'll find sooo many tweets where people have expressed how much they have learnt CSS by playing here. I'll explain how: 
1. even though the code you write here is unprofessional and hacky, the intuitive understanding you build in CSS properties while solving the targets here is unparallel.
2. Using those properties number of times daily gives you a practice on properties/values like no other place!
3. You would develop your creative thinking when trying to figure out different approaches for each of the targets which again helps when developing professional UI layouts with CSS.
4. And while you are doing all this, you get to use so many CSS properties which you wouldn't otherwise get to use normally in professional domain and that broadens one's toolset making one more capable of using the best property for the work at hand.
5. Last but not the least, having some fun with CSS makes you fall in ❤️ with CSS more! :)

In short, once you understand what you gain at CSSBattle, it's a place to be!
That said, we respect your viewpoint about the platform, whatever you decide at the end.

**Comment by ParrhesiaJoe** [comment]
Is the code used by the winners shared? I assumed I could look at the winning examples and learn some tricks from that, but I couldn't see anyone else’s code for the battles, so there was nothing to learn from.

Where do I see the code that will teach me?

 

Joe

**Reply by chinchang** [reply]
We launched Daily targets just for that - https://cssbattle.dev/daily
One new target launches daily - no leaderboard, no competition. Next day its top 10 solution become public for everyone to learn from.
Plus, our discord server has very interesting discussions on CSS properties, approaches and more :) - https://discord.gg/rvqvbcFM86

**Reply by muhamed-kanapiya** [reply]
❤️

**Comment by g-schaffer** [comment]
I totally agree with [ParrhesiaJoe](https://github.com/ParrhesiaJoe), adding a "minify" button that removes spaces would be a nice idea (and potentially a "unminify" button to revert it)

**Reply by ngdangtu-vn** [reply]
I think they have to minify btn for premium users. Personally, I'm fine with DDG minifier :))

**Comment by muhamed-kanapiya** [comment]
For me its fun and helps to train CSS like algos form leetcode

**Comment by sb1701e** [comment]
When I first attempt a problem, I do so using more properly written CSS, and only once I have 100% do I manually tweak things to reduce the number of characters used. I find the whole process a great way to learn and experiment with different features of CSS. One area I have really gained practice is with the various background options, as several puzzles, which at first seem like would require many elements can be achieved with with the use one or two backgrounds. Another CSS trick I was happy to find a use for was mix-blend-mode in puzzle #32, but that is the only time I've been able to use that. Could I learn like this outside of CSS battle, sure, but this sites makes learning fun for me.
